src/OSPSuite.Core/Domain/ObjectSource.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Visitor;
+
+namespace OSPSuite.Core.Domain
+{
+   /// <summary>
+   ///    Tracks the origin of specific entities in the simulation (such as parameters, initial conditions, parameter values
+   ///    and formulas).
+   ///    The origin module will be the one that last modified the entity (e.g a parameter might be created by multiple
+   ///    module. This will be the one that created the entity)
+   ///    Each source is unique by path. It will store the id of the module, potentially the id of the building block, the
+   ///    type of the building block
+   /// </summary>
+   public class ObjectSource
+   {
+      /// <summary>
+      ///    Id of the object in the simulation.
+      /// </summary>
+      public string ObjectId { get; }
+
+      public string ModuleId { get; }
```

Comment:
I decided to not track an path in this object. Only id. This will be serialized and saved in the simulation. That way, we don't have to deal with rename etc.. When we present, we can always retrieve path, module name etc from id